Kludex/uvicorn
# chore/get commits to release

- Fix PEP 508 violation (#1035)
- Moved FlowControl class and globals to its own file in uvicorn/protocols/http/ (#1034)
- 📝 Update markdown extensions to highlight code (#1043)
- ♻️ Replace _types module by asgiref.typing (#1044)
- Change websockets version to v9 (#1038)
- Improve user feedback if no ws library installed (#1023)
- Update docs about reload_dirs (#1028)
- Add docs warning about reload and debug flags (#949) (#1045) (#1047)
- Fix gunicorn worker not running if extras not installed (#901)
- update httptools from 0.1.* to 0.2.* (#1024)
- tests: ignore coverage on helper test functions (#980)
- coverage: pin higher coverage (#1006)
- fix: add mypy on basereload.py (#994)
- Defaults ws max_size on server to 16MB (#995)
- fix: add mypy on asgi2.py (#1000)
- fix: add mypy on logging.py (#1003)
- fix: add mypy on watchgodreload.py (#993)
- Test logging (#964)
- fix: add mypy on debug.py (#1002)
- fix: mypy issues (#990)
- fix: add mypy on message_logger.py (#1004)
- Switch to asyncio streams API (#869)
- Improved wsgi middleware coverage (#1050)
- tests: add env file config coverage (#985)
- 🚨 Cover mypy on  _handlers subpackage (#1051)
- Fixed typo reload vs reloader (#971)
- 🚨 Cover __main__.py on mypy (#1052)
- 🚨 Cover __init__.py on mypy (#1055)
- 🚨 Cover subprocess.py on mypy (#1053)
- Coverage logging (#1057)
- 📝 Remove community discussion link (#1063)
- Force reload_dirs to be a list (#978)
- 🚨 Cover supervisors/multiprocess.py on mypy (#1059)
- 🚨 Cover supervisors/statreload.py on mypy (#1060)
- 🚨 Cover protocols/http/flow_control.py on mypy (#1061)
- Upgraded websockets requirements (#1065)
- Strict pytest (#929)
- 0.14.0 (#1056)
- Add option to set websocket ping interval and timeout (#1048)
- [Docs] Add uvloop and httptools related info (#1071)
- ➕ Add types-click (#1072)
- 🚨 Cover init files on mypy (#1074)
- Support disabling default Server and Date headers (#818)
- Correct usage of param "message" in logging (#1082)
- Revert propagate status after exiting context caplog_for_logger() (#1086)
- Add missing trace log for websocket protocols (#1083)
- Removed print from test (#1087)
- Add parametrized http protocols on ws tests (#1080)
- Moved TRACE_LOG_LEVEL in one spot (#1084)
- Added existence check to the reload directory(ies) (#1089)
- Type-annotate uvicorn/config.py (#1067)
- Adapt bind_socket to make it usable with multiple processes (#1009)
- Exit with status 3 when worker starts failed (#1077)
- 🚨 Cover middleware/wsgi.py on mypy (#1075)
- fix: add mypy on proxy_headers.py (#999)
- 🚨 Cover server.py on mypy (#1054)
- 🔧 Add funding option (#1097)
- :arrow_up: Bump asgiref to 3.4.0 (#1100)
- :package: Add PEP440 compliant version of click (#1099)
- :hammer: Add script that fills CHANGELOG to release
